apps/pay/utils/utils.ts

```diff
@@ -120,3 +123,7 @@ export const getLocaleConfig = (intlConfig?: IntlConfig): LocaleConfig => {
     return prev
   }, defaultConfig)
 }
+
+export const extractSearchParams = (searchParams: ReadonlyURLSearchParams) => {
+  return searchParams ? Object.fromEntries(searchParams.entries()) : {}
+}
```

Comment:
I still think that using `searchParams.get()` is better than creating a this util function.

```
  const searchParams = useSearchParams()
  const { username } = useParams()
  const query = extractSearchParams(searchParams)
  ```
  
  this code is convoluted, and require more digging than just doing 

```
const searchParams = useSearchParams()
const amount = searchParams.get('amount')
```

additionaly, I think using https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams instead of play javascript when crafting new URL is a way to prevent edge cases (I used to do that before, but there were all case of side issues that was prevent by the use of the proper API, ie: https://github.com/GaloyMoney/galoy-mobile/pull/3070/files)

Reply:
and if this needs transformation, this can be used like this:

```
const params = new URLSearchParams(searchParams.toString());
```

like as an example here: https://nextjs.org/blog/next-14-1#windowhistorypushstate-and-windowhistoryreplacestate

Reply:
> I still think that using `searchParams.get()` is better than creating a this util function.
> 
> ```
>   const searchParams = useSearchParams()
>   const { username } = useParams()
>   const query = extractSearchParams(searchParams)
> ```
> 
> this code is convoluted, and require more digging than just doing
> 
> ```
> const searchParams = useSearchParams()
> const amount = searchParams.get('amount')
> ```
> 
> additionaly, I think using https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams instead of play javascript when crafting new URL is a way to prevent edge cases (I used to do that before, but there were all case of side issues that was prevent by the use of the proper API, ie: https://github.com/GaloyMoney/galoy-mobile/pull/3070/files)

Okay, I will make changes, but in some places, we are required to use an object of params. There, we can use this for now, but when we do UI refactoring, we can remove this in those places as well because we are going to remove some UI elements, and I think it would be easier to do refactoring then. Object of params is required because the older `router` used to provide params as an object via `router.query`.